### PR TITLE
Mention ccache in developer docs

### PR DESCRIPTION
**Pull request by Alexander-Wilms**
Or should ENABLE_CCACHE be OFF by default?

### CODE REVIEW

File: docs/developers/Building_macOS.md [hunk 1]
```diff
@@ -7,7 +7,9 @@
     - Xcode IDE: <https://developer.apple.com/xcode/>
     - (not tested) other C++ compilers, e.g. gcc/clang from [Homebrew](https://brew.sh/)
 2. CMake: `brew install --cask cmake` or get from <https://cmake.org/download/>
-3. (optional) Ninja: `brew install ninja` or get from <https://github.com/ninja-build/ninja/releases>
+3. CCache to speed up recompilation: `brew install ccache`
```
Comment (kambala-decapitator) [inline docs/developers/Building_macOS.md:10]: this should also be optional

**Comment by kambala-decapitator** [comment]
why did you decide not to edit android and iOS docs?

> Or should ENABLE_CCACHE be OFF by default?

yes, I guess so

File: docs/developers/Building_Android.md [hunk 2]
```diff
@@ -14,7 +14,9 @@ The following instructions apply to **v1.2 and later**. For earlier versions the
     - install with `sdkmanager` command line tool
     - download from https://developer.android.com/ndk/downloads
     - download with Conan, see [#NDK and Conan](#ndk-and-conan)
-5. (optional) Ninja: download from your package manager or from https://github.com/ninja-build/ninja/releases
+5. Optionsal:
```
Comment (kambala-decapitator) [inline docs/developers/Building_Android.md:17]:
```suggestion
5. Optional:
```

File: docs/developers/Building_Windows.md [hunk 3]
```diff
@@ -10,7 +10,10 @@ Windows builds can be made in more than one way and with more than one tool. Thi
 - Git or git GUI, for example, SourceTree [download link](http://www.sourcetreeapp.com/download)
 - CMake [download link](https://cmake.org/download/). During install after accepting license agreement make sure to check "Add CMake to the system PATH for all users".
 - To unpack pre-build Vcpkg: [7-zip](http://www.7-zip.org/download.html)
-- Optionally, to create installer: [NSIS](http://nsis.sourceforge.net/Main_Page)
+- To speed up recompilation: [CCache](https://github.com/ccache/ccache/releases)
```
Comment (kambala-decapitator) [inline docs/developers/Building_Windows.md:13]: That's also optional

File: docs/developers/Building_Windows.md [hunk 4]
```diff
@@ -75,6 +78,10 @@ From command line use:
 
 For the list of the packages used you can also consult [vcmi-deps-windows readme](https://github.com/vcmi/vcmi-deps-windows) in case this article gets outdated a bit.
 
+# Install CCache
+
+Extract `ccache` to a folder of your choosing, add the folder to the `PATH` environment variable and log out and back in.
+
```
Comment (kambala-decapitator) [inline docs/developers/Building_Windows.md:84]: Don't you need to enable our option for msvc?

File: docs/developers/Building_Windows.md (hunk 3 above)
Comment (Alexander-Wilms) [reply docs/developers/Building_Windows.md:13]: I'll need to find out how to use CMake variables when building with Visual Studio

**Pull request merged**